gratipay/utils/__init__.py

```diff
@@ -132,6 +133,46 @@ def get_team(state):
     return team
 
 
+def b64encode_s(s):
+    prefix = b''
+    if not isinstance(s, bytes):
+        s = s.encode('utf8')
+    else:
+        # Check whether the string is binary or already utf8
+        try:
+            s.decode('utf8')
+        except UnicodeError:
+            prefix = b'.'
+    return prefix + b64encode(s, b'-_').replace(b'=', b'~')
+
+
+def b64decode_s(s, **kw):
+    def error():
+        if 'default' in kw:
+            return kw['default']
+        raise Response(400, "invalid base64 input")
+
+    try:
+        s = s.encode('ascii')
+    except UnicodeError:
+        return error()
+
+    udecode = lambda a: a.decode('utf8')
+    if s[:1] == b'.':
+        udecode = lambda a: a
+        s = s[1:]
+    s = s.replace(b'~', b'=')
+    try:
+        return udecode(b64decode(s, '-_'))
+    except Exception:
+        try:
+            # For retrocompatibility
+            return udecode(b64decode(s))
+        except Exception:
+            pass
+        return error()
+
+
```

Comment:
> bytes/unicode conversion.
> . prefix

I don't like that we're so relaxed wrt bytes and unicode here. Seems like a slide back to Python 2 ways of thinking. Why do we need to accept (in `b64encode_s`) and return (from `b64decode_s`) bytes? Why not only accept and return unicode? Our use case for these functions is to transport strings to ourselves across a querystring. We're not trying to send images across the querystring. I propose that we rename the functions to `pack_for_querystring` and `unpack_from_querystring`, and that we constrain them to accept and return only unicodes.